tasks/build/download_node_builds.js

```diff
@@ -1,27 +1,95 @@
 module.exports = function (grunt) {
   let { map, fromNode } = require('bluebird');
-  let { resolve } = require('path');
-  let { pluck } = require('lodash');
-  let { createWriteStream } = require('fs');
+  let { resolve, basename } = require('path');
+  let { createReadStream, createWriteStream } = require('fs');
   let { createGunzip } = require('zlib');
   let { Extract } = require('tar');
   let { rename } = require('fs');
+  let checkHash  = require('check-hash');
   let wreck = require('wreck');
 
-  let platforms = grunt.config.get('platforms');
-  let activeDownloads = [];
+  const platforms = grunt.config.get('platforms');
+  let shaSums = {};
+  const getShaSums = async (done) => {
+    const nodeVersion = grunt.config.get('nodeVersion');
+    const shaSumsUri = `https://nodejs.org/dist/v${nodeVersion}/SHASUMS256.txt`;
 
-  let start = async (platform) => {
-    let finalDir = platform.nodeDir;
-    let downloadDir = `${finalDir}.temp`;
+    const resp = await fromNode(cb => {
+      wreck.get(shaSumsUri, function (err, resp, payload) {
+        if (err) {
+          return cb(err);
+        }
+
+        if (resp.statusCode !== 200) {
+          return cb(new Error(`${shaSumsUri} failed with a ${resp.statusCode} response`));
+        }
+
+        cb(null, payload);
+      });
+    });
+
+    resp.toString('utf8').split('\n').forEach(line => {
+      let cells = line.split('  ');
+      shaSums[cells[1]] = cells[0];
+    });
+
+    grunt.log.ok(`sha sums download complete`);
+    done();
+  };
+
+  const checkShaSum = async (platform) => {
+    const finalDir = platform.nodeDir;
+    const file = basename(platform.nodeUrl);
+    const path = resolve(platform.nodeDir, file);
+    const expected = {
+      hash: 'sha256',
+      expected: platform.win ? shaSums['win-x86/' + file] : shaSums[file]
+    };
+    const resp = await fromNode(cb => {
+      checkHash.fromFile(path, expected, function (err, passed, actual) {
+        if (err) {
+          cb(err);
+        } else if (!passed) {
+          grunt.log.fail(`${platform.name} hash is not valid: ${expected.expected} :: ${actual}`);
+          cb(null, false);
+        } else {
+          grunt.log.ok(`${platform.name} shasum verified`);
+          if (!platform.win) {
+            createReadStream(path)
+              .pipe(createGunzip())
+              .on('error', (err) => {
+                cb(new Error(`error extracting: ${err.message}`));
+              })
+              .pipe(new Extract({path: platform.nodeDir, strip: 1}))
+              .on('error', (err) => {
+                cb(new Error(`error extracting: ${err.message}`));
+              })
+              .on('end', () => {
+                cb(null, true);
+              });
+          }
+        }
+      });
+    });
+    if (resp !== true) grunt.file.delete(finalDir);
```

Comment:
Same comment as above, it seems like it would make more sense to pull this out into the function that calls `checkShaSum`.